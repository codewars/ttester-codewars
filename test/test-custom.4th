\ Copyright 2019 nomennescio

s" custom messages" describe#{

  :noname ." Just passed" cr ; ^passed$ !
  :noname ." Just failed" cr ; ^different$ !

  s" short strings" it#{
    <{ s" Hello World!" s# -> s" Hello World!" s# }>
    <{ s" Hello Worlds!" s# -> s" Hello Worlds!" s# }>
  }#
  s" failing compares" it#{
    <{ s" Hello World!" s# -> s" Hello Worlds!" s# }>
    <{ s" Hello Worlds!" s# -> s" Hello Worlds! " s# }>
  }#

  2variable actual$
  2variable expected$

  :noname ." Got '" actual$ 2@ type ." ' as expected" cr ; ^passed$ !
  :noname ." Expected '" expected$ 2@ type ." ', got '" actual$ 2@ type ." '" cr ; ^different$ !

  : &actual 2dup actual$ 2! ;
  : &expected 2dup expected$ 2! ;

  s" short strings" it#{
    <{ s" Hello World!" &actual s# -> s" Hello World!" &expected s# }>
    <{ s" Hello Worlds!" &actual s# -> s" Hello Worlds!" &expected s# }>
  }#
  s" failing compares" it#{
    <{ s" Hello World!" &actual s# -> s" Hello Worlds!" &expected s# }>
    <{ s" Hello Worlds!" &actual s# -> s" Hello Worlds! " &expected s# }>
  }#

}#
