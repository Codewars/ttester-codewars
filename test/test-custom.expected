
<DESCRIBE::>custom messages

<IT::>short strings

<PASSED::>Just passed

<PASSED::>Just passed

<COMPLETEDIN::>0.139 ms

<IT::>failing compares

<FAILED::>Just failed

<FAILED::>Just failed

<COMPLETEDIN::>0.162 ms

<IT::>short strings

<PASSED::>Got 'Hello World!' as expected

<PASSED::>Got 'Hello Worlds!' as expected

<COMPLETEDIN::>0.149 ms

<IT::>failing compares

<FAILED::>Expected 'Hello Worlds!', got 'Hello World!'

<FAILED::>Expected 'Hello Worlds! ', got 'Hello Worlds!'

<COMPLETEDIN::>0.150 ms

<COMPLETEDIN::>1.365 ms
